adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -168,11 +169,58 @@ public void run() {
 
                     final AcquireTokenSilentHandler acquireTokenSilentHandler = new AcquireTokenSilentHandler(mContext,
                             authenticationRequest, mTokenCacheAccessor);
-                    final AuthenticationResult authResult
+                    final AuthenticationResult result
                             = acquireTokenSilentHandler.acquireTokenWithRefreshToken(refreshToken);
+
+                    final String correlationId = String.format(" CorrelationId: %s", authenticationRequest.getCorrelationId().toString());
+
+                    if (result == null) {
+                        Logger.e(TAG + methodName, "Returned result with exchanging refresh token for access token is null" + correlationId, "",
+                                ADALError.AUTH_REFRESH_FAILED);
+                        throw new AuthenticationException(
+                                ADALError.AUTH_REFRESH_FAILED, correlationId);
```

Comment:
Do we want to include some additional message here that _nothing_ was received?

Reply:
Addressed